public/scripts/project/remote.js

```diff
@@ -0,0 +1,142 @@
+define(function(require) {
+  var $ = require("jquery");
+
+  var Path = Bramble.Filer.Path;
+  var Buffer = Bramble.Filer.Buffer;
+
+  // Installs a tarball (arraybuffer) containing the project's files/folders.
+  function installTarball(fs, root, tarball, callback) {
+    var untarWorker;
+    var pending = null;
+    var sh = new fs.Shell();
+
+    function extract(path, data, callback) {
+      path = Path.resolve(root, path);
```

Comment:
This won't work. `Path.resolve("/1/projects/5", "/index.html")` results in `/index.html`. Instead, we should probably just follow the convention that we are sending `/index.html` and instead just call `addRoot(path)`.